db-server/test/backend/db_tests.js

```diff
@@ -2147,6 +2155,73 @@ module.exports = function (config, DB) {
       })
     })
 
+    describe('account recovery key', () => {
+      let account, data
+      beforeEach(() => {
+        account = createAccount()
+        account.emailVerified = true
+        return db.createAccount(account.uid, account)
+          .then(() => {
+            data = createRecoveryData()
+            return db.createRecoveryKey(account.uid, data)
+          })
+          .then((res) => {
+            assert.ok(res, 'empty response')
+          })
+      })
+
+      it('should fail to create for unknown user', () => {
+        return db.createRecoveryKey('12312312312', data)
+          .then(assert.fail, (err) => {
+            assert.equal(err.errno, 116, 'not found')
+          })
+      })
+
+      it('should create account recovery key', () => {
+        data = createRecoveryData()
+        return db.createRecoveryKey(account.uid, data)
+          .then((res) => {
+            assert.ok(res, 'empty response')
+          })
+      })
+
+      it('should fail to create multiple keys', () => {
+        return db.createRecoveryKey(account.uid, data)
```

Comment:
This tests that you can't created two keys with the same id, but not that you can only have one key on the account.